### PR TITLE
[DOC release] updated docs regarding ember debug methods

### PR DESCRIPTION
**Pull request by webark**
This brings them all in line with #13180

Was brought up during conversation in https://github.com/ember-cli/rfcs/pull/50#issuecomment-206509353

### CODE REVIEW

File: packages/ember-debug/lib/deprecate.js [hunk 1]
```diff
@@ -95,8 +95,9 @@ export let missingOptionsUntilDeprecation = 'When calling `Ember.deprecate` you
 
 /**
   Display a deprecation warning with the provided message and a stack trace
-  (Chrome and Firefox only). Ember build tools will remove any calls to
-  `Ember.deprecate()` when doing a production build.
+  (Chrome and Firefox only). Ember build tools will remove any calls
+  to `Ember.deprecate()` when doing an Ember.js framework production build
+  and will make the assertion a no-op for an application production build.
```
Comment (justinaray) [inline packages/ember-debug/lib/deprecate.js]: Is `make the assertion` a copy/paste mistake from the assert jsdoc?

Comment (mixonic) [inline packages/ember-debug/lib/deprecate.js]: Seems like it should say `deprecation`. Is that really the app-space behavior for deprecate in prod?

File: packages/ember-debug/lib/index.js [hunk 2]
```diff
@@ -102,8 +105,9 @@ setDebugFunction('info', function info() {
   Display a deprecation warning with the provided message and a stack trace
   (Chrome and Firefox only) when the assigned method is called.
 
-  Ember build tools will not remove calls to `Ember.deprecateFunc()`, though
-  no warnings will be shown in production.
+  Ember build tools will remove any calls to `Ember.deprecateFunc()` when
+  doing an Ember.js framework production build and will make the assertion a
+  no-op for an application production build.
```
Comment (mixonic) [inline packages/ember-debug/lib/index.js]: This is not true. The original message seems correct?

File: packages/ember-debug/lib/warn.js [hunk 3]
```diff
@@ -24,8 +24,9 @@ export let missingOptionsIdDeprecation = 'When calling `Ember.warn` you must pro
 */
 
 /**
-  Display a warning with the provided message. Ember build tools will
-  remove any calls to `Ember.warn()` when doing a production build.
+  Display a warning with the provided message. Ember build tools will remove
+  any calls to `Ember.warn()` when doing an Ember.js framework production build
+  and will make the assertion a no-op for an application production build.
```
Comment (mixonic) [inline packages/ember-debug/lib/warn.js]: Not an assertion.

File: packages/ember-debug/lib/deprecate.js (hunk 1 above)
Comment (justinaray) [inline packages/ember-debug/lib/deprecate.js]: I cannot speak for @webark, but I think this was to eventually match the behavior to be implemented by https://github.com/ember-cli/rfcs/pull/50.

We recently came across this when we removed all custom deprecation handlers and the default Ember console.log implementation started to flood our prod consoles.

I took from my slack conversation that this RFC was to land quickly.  Which might be the motivation not to updated the docs to match current behavior only to change them back after impl?

**Comment by webark** [comment]
Ya, I was simply copying and pasting from what was decided in #13180 and seems like some better wording should be done. The motivation for this PR was two things. First, based off of the current documentation, I thought that `Ember.info` would behave the same in production, cause it never said elsewise in the documentation. Second, it does seem based off of the wording that it will actually remove the calls, not just turn them into a empty function.

**Comment by webark** [comment]
I'll update the wording for them though so that it's a little better then a :scissors: :honey_pot: (didn't have a brush or glue. figure honey could be used as glue in a pinch)

**Comment by webark** [comment]
was just basing which methods where 'no-op'ed from this. https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/debug.js I left out the debug seal cause There was no documentation on that as is.

File: packages/ember-debug/lib/deprecate.js (hunk 1 above)
Comment (mixonic) [inline packages/ember-debug/lib/deprecate.js]: We should definitely not add documentation that is incorrect, but would be if/when some other work lands. Right?

**Comment by mixonic** [comment]
Seems totally great to move forward with! We should just be sure the docs are accurate :-)

**Comment by webark** [comment]
@mixonic @justinaray Updated to hopefully be more strait forward and less wordy. Once ember-cli/rfcs#50 is complete, the documentation could be updated to reflect the eventual implementation.

File: packages/ember-debug/lib/deprecate.js [hunk 4]
```diff
@@ -95,8 +95,9 @@ export let missingOptionsUntilDeprecation = 'When calling `Ember.deprecate` you
 
 /**
   Display a deprecation warning with the provided message and a stack trace
-  (Chrome and Firefox only). Ember build tools will remove any calls to
-  `Ember.deprecate()` when doing a production build.
+  (Chrome and Firefox only).
+
+  * In a production build, this method is defined as an empty function (NOP).
```
Comment (mixonic) [inline packages/ember-debug/lib/deprecate.js]: Can we note both behaviors?

> * In a production build, this method is defined as an empty function (NOP). Uses of this method in Ember itself are stripped from the `ember.prod.js` build.

File: packages/ember-debug/lib/index.js [hunk 5]
```diff
@@ -30,10 +30,9 @@ import { generateTestAsFunctionDeprecation } from 'ember-debug/handlers';
 
 
 /**
-  Define an assertion that will throw an exception if the condition is not
-  met. Ember build tools will remove any calls to `Ember.assert()` when
-  doing an Ember.js framework production build and will make the assertion a
-  no-op for an application production build. Example:
+  Define an assertion that will throw an exception if the condition is not met.
+
+  * In a production build, this method is defined as an empty function (NOP).
```
Comment (mixonic) [inline packages/ember-debug/lib/index.js]: Can we note both behaviors?

> * In a production build, this method is defined as an empty function (NOP). Uses of this method in Ember itself are stripped from the `ember.prod.js` build.

**Comment by mixonic** [comment]
I've noted two spots where we should mention the calls being stripped from production, but you should look to the actual list for confirmation:

* https://github.com/emberjs/ember.js/blob/master/ember-cli-build.js#L38

Thanks! :clap:

**Comment by webark** [comment]
@mixonic updated! And no worries! 😄

**Comment by mixonic** [comment]
Thanks @webark!

**Pull request merged**